apps/avifenc.c

```diff
@@ -1475,14 +1475,13 @@ int main(int argc, char * argv[])
     // Check lossy/lossless parameters and set to default if needed.
     if (lossless) {
         // Pixel format.
-        if (input.requestedFormat != AVIF_PIXEL_FORMAT_NONE && input.requestedFormat != AVIF_PIXEL_FORMAT_YUV444) {
+        if (input.requestedFormat != AVIF_PIXEL_FORMAT_NONE && input.requestedFormat != AVIF_PIXEL_FORMAT_YUV444 &&
+            input.requestedFormat != AVIF_PIXEL_FORMAT_YUV400) {
             fprintf(stderr,
                     "When set, the pixel format can only be 444 in lossless "
-                    "mode.\n");
+                    "mode. 400 also works if the input is grayscale.\n");
```

Comment:
Ack. I also considered that an image stored as RGB but asked to be compressed losslessly using 4:0:0 does not work because it is considered as lossy. Then again, if your original image is grayscale, there will indeed be no visual loss. BUT, instead of 3 channels, you'll have one. I though that was enough to consider the images different, hence lossy.

Reply:
There is a technical reason to avoid using CICP=1/13/0 with lossless 4:0:0. See the following excerpt of the color_config() function in Section 5.5.2 (Color config syntax) of the AV1 spec:

```
    if ( mono_chrome ) {
        color_range                                            f(1)
        subsampling_x = 1
        subsampling_y = 1
        chroma_sample_position = CSP_UNKNOWN
        separate_uv_delta_q = 0
        return
    } else if ( color_primaries == CP_BT_709 &&
                transfer_characteristics == TC_SRGB &&
                matrix_coefficients == MC_IDENTITY ) {
        color_range = 1
        subsampling_x = 0
        subsampling_y = 0
    } else {
```

If `mono_chrome` is true, then AV1 specifies that `subsampling_x` and `subsampling_y` be equal to 1.

The "else if" statement means "else if CICP == 1/13/0". Then it specifies that `subsampling_x` and `subsampling_y` be equal to 0.

So if both `mono_chrome` is equal to 1 and CICP == 1/13/0, then the AV1 spec appears to require conflicting values for `subsampling_x` and `subsampling_y`.

Reply:
More on the technical reason: Section 6.4.2 (Color config semantics) of the AV1 spec also says, on page 119:

> If matrix_coefficients is equal to MC_IDENTITY, it is a requirement of bitstream conformance that subsampling_x is equal to 0 and subsampling_y is equal to 0.

Since `subsampling_x` is equal to 1 and `subsampling_y` is equal to 1 if `mono_chrome` is equal to 1, `matrix_coefficients == MC_IDENTITY` seems incompatible with `mono_chrome == 1`.